examples/channels-example/frontend/src/components/Channels/Channels.jsx

```diff
@@ -0,0 +1,174 @@
+import React, { PureComponent } from 'react';
+import ws from '../../utils/ws';
+import sse from '../../utils/sse';
+import { PRIVILEGED_ROLES } from '../../constants';
+import Events from './Events';
+
+class Channels extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      protocols: { ws, sse },
+      type: 'ws',
+      username: '',
+      levels: '',
+      subscriberTopic: '',
+      status: 'None',
+      response: '',
+      messages: []
+    };
+
+    this.connectToChannel = this.connectToChannel.bind(this);
```

Comment:
just a remark: if instead of `connectToChannel() {` you use `connectToChannel = () => {`, you can omit (read: can't forget) the explicit call to bind. I tend to use this consistently for all methods that can be used as a callback.